File: Languages/Java/Day012.java
// Languages - Java - Day 12
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that takes two numbers as parameters and returns the larger of the two. Call this function and print the result.

import java.util.Scanner;

public class Day012 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Get two valid numbers from the user
        int num1 = getValidNumber(scanner, "Enter the first number: ");
        int num2 = getValidNumber(scanner, "Enter the second number: ");

        // Find the larger number
        int largerNumber = findLarger(num1, num2);

        // Print the result
        System.out.println("The larger number is: " + largerNumber);
        scanner.close();
    }

    // Function to find the larger of two numbers
    public static int findLarger(int a, int b) {
        return Math.max(a, b);
    }

    // Function to get a valid integer input from the user
    public static int getValidNumber(Scanner scanner, String prompt) {
        while (true) {
            try {
                System.out.print(prompt);
                return Integer.parseInt(scanner.nextLine());
            } catch (NumberFormatException e) {
                System.out.println("Invalid input. Please enter a valid integer.");
            }
        }
    }
}

File: Languages/Kotlin/Day012.kt
// Languages - Kotlin - Day 12
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that takes two numbers as parameters and returns the larger of the two. Call this function and print the result.

import java.util.Scanner

// Function to find the larger of two numbers
fun findLarger(num1: Double, num2: Double): Double {
    return if (num1 > num2) num1 else num2
}

// Function to get a valid number from the user
fun getValidNumber(prompt: String): Double {
    val scanner = Scanner(System.`in`)
    while (true) {
        print(prompt)
        val input = scanner.nextLine()
        try {
            return input.toDouble()
        } catch (e: NumberFormatException) {
            println("Invalid input. Please enter a valid number.")
        }
    }
}

fun main() {
    // Get two valid numbers from the user
    val num1 = getValidNumber("Enter the first number: ")
    val num2 = getValidNumber("Enter the second number: ")

    // Call the function and print the result
    val largerNumber = findLarger(num1, num2)
    println("The larger number is: $largerNumber")
}
